more fixes for charcoal

temp1011 opened the pull request:
- use null instead of air blocks in CharcoalPileWall to prevent matching returning true for air blocks:
Caused a regression where charcoal pits with invalid walls would just match the first wall because air state would match with any walls registered with just a block.

- don't give any charcoal for blocks that aren't charcoal walls:
The old code (#2241 and before) gave at least 1 charcoal per face but I don't think this is balanced, since charcoal piles then just become a furnace for free that give ash too. Also updates `BlockAsh` to use the new `getDrops` method.

Nedelosk commented on src/main/java/forestry/arboriculture/blocks/BlockWoodPile.java:
```diff
@@ -201,7 +201,11 @@ private float getCharcoalAmount(World world, BlockPos pos) {
-		return charcoalAmount / 6;
+		float ret = charcoalAmount / 6;
+		if (ret > 0F) {
+			ret += 9F;    //base charcoal drop amount for ash block, calculated here instead so invalid walls son't still drop charcoal
```
What if the ret is at this moment bigger than 6? The ash block only has a valid metadata from 0 to 15.

temp1011 replied:
I'm not sure what would happen but it probably wouldn't be good. Hopefully fixed.

temp1011 committed:
more fixes for charcoal

	- use null instead of air blocks in CharcoalPileWall to prevent matching returning true for air blocks
	- don't give any charcoal for blocks that aren't charcoal walls

temp1011 committed:
fix typo

Nedelosk commented on src/main/java/forestry/arboriculture/blocks/BlockWoodPile.java:
```diff
@@ -201,7 +201,11 @@ private float getCharcoalAmount(World world, BlockPos pos) {
-		return charcoalAmount / 6;
+		float ret = charcoalAmount / 6;
+		if (ret > 0F) {
+			ret += 9F;    //base charcoal drop amount for ash block, calculated here instead so invalid walls won't still drop charcoal
```
Why is this at `BlockWoodPile.getCharcoalAmount` and not in `BlockAsh.getDrops` after `if (amount > 0)` ?

temp1011 replied:
Not sure, probably I got confused. Having it in this method changes behaviour too because the amount gets clamped at the end.

temp1011 committed:
simplify things
